src/CampaignKit.WorldMap/Controllers/MapController.cs

```diff
@@ -180,279 +151,348 @@ public async Task<IActionResult> Create(MapCreateViewModel model)
                 Id = id,
-                CanEdit = canEdit
+                CanEdit = canEdit,
             };
 
-            ViewBag.MaxZoomLevel = map.MaxZoomLevel;
-            ViewBag.WorldPath = Url.Content($"{_filePathService.VirtualWorldBasePath}/{id}");
-            ViewBag.NoWrap = !map.RepeatMapInX;
+            this.ViewBag.MaxZoomLevel = map.MaxZoomLevel;
+            this.ViewBag.WorldPath = $"{this.configuration.GetValue<string>("AzureBlobBaseURL")}/map{map.MapId}";
+            this.ViewBag.NoWrap = !map.RepeatMapInX;
 
-            return View(model);
+            return this.View(model);
         }
 
-        #endregion
-
-        #region  Marker Related Actions    
-
         /// <summary>
-        ///     GET: /Map/MarkerData/{id?}
+        ///     GET: /Map/MarkerData/{id?}.
         /// </summary>
         /// <param name="id">The map identifier.</param>
         /// <param name="shareKey">The map's shareKey.</param>
         /// <returns>The map's marker data in JSON format.</returns>
-        [HttpGet]
-        public async Task<IActionResult> MarkerData(int id, string shareKey)
+        [HttpGet("MarkerData/{id?}")]
+        public async Task<IActionResult> MarkerData(string id, string shareKey)
         {
             // Determine if user can view map
-            var canView = await _mapRepository.CanView(id, User, shareKey);
-            if (!canView) return ShowErrorView();
+            var canView = await this.mapRepository.CanView(id, this.User, shareKey);
+            if (!canView)
+            {
+                return this.ShowErrorView();
+            }
 
             // Retrieve the map
-            var map = await _mapRepository.Find(id, User, shareKey);
+            var map = await this.mapRepository.Find(id, this.User, shareKey);
 
             // Create response data
             var markerData = map.MarkerData;
-            if (string.IsNullOrEmpty(markerData)) markerData = "[]";
-            return Json(markerData);
+            if (string.IsNullOrEmpty(markerData))
+            {
+                markerData = "[]";
+            }
+
+            return this.Json(markerData);
         }
 
         /// <summary>
-        ///     POST: Map/MarkerData/{MapId}
+        ///     POST: Map/MarkerData/{MapId}.
         /// </summary>
         /// <param name="model">Map marker data.</param>
-        /// <returns></returns>
-        [HttpPost]
+        /// <returns>Response JSON.</returns>
+        [HttpPost("MarkerData/{MapId?}")]
         [Authorize]
         public async Task<IActionResult> MarkerData([FromBody] MarkerEditViewModel model)
         {
             // Ensure that user has privileges to update map
-            var canEdit = await _mapRepository.CanEdit(model.MapId, User);
-            if (!canEdit) return Json("Not Authorized");
+            var canEdit = await this.mapRepository.CanEdit(model.MapId, this.User);
```

Comment:
This combination of route param and [FromBody] I do not get - will map id be read from the URL or from the body content?

Reply:
This is not working correctly.  I am looking into it.

Reply:
Still having problems with this since I removed the {id} param.  I'm going to use Postman to diagnose.

Reply:
This has been corrected.  There was no need for the route param.